e2e/LongHaul/amqpPooling/IotHub.cs

```diff
@@ -0,0 +1,192 @@
+
+            s_devices = devices;
+            s_deviceClients = new Dictionary<string, IotHubDeviceClient>();
+        }
+
+        public Dictionary<string, string> TelemetryUserProperties { get; } = new();
+
+        public async Task InitializeAsync()
+        {
+            await _lifetimeControl.WaitAsync().ConfigureAwait(false);
+
+            var helper = new IotHubConnectionStringHelper(_hubConnectionString);
+
+            try
+            {
+                foreach (var device in s_devices)
+                {
+                    string deviceConnectionString = $"HostName={helper.HostName};DeviceId={device.Id};SharedAccessKey={device.Authentication.SymmetricKey.PrimaryKey}";
+
+                    var deviceClient = new IotHubDeviceClient(deviceConnectionString, _clientOptions);
+
+                    // Set the device client connection status change handler
+                    var amqpConnectionStatusChange = new AmqpConnectionStatusChange(_logger.Clone());
+                    deviceClient.ConnectionStatusChangeCallback = amqpConnectionStatusChange.ConnectionStatusChangeHandler;
+
+                    await deviceClient.OpenAsync().ConfigureAwait(false);
+
+                    s_deviceClients.Add(device.Id, deviceClient);
+                }
+            }
+            finally
+            {
+                _lifetimeControl.Release();
+            }
+        }
+
+        public async Task SendTelemetryMessagesAsync(CancellationToken ct)
+        {
+            var sw = new Stopwatch();
+
+            while (!ct.IsCancellationRequested)
+            {
+                await Task.Delay(s_messageLoopSleepTime, ct).ConfigureAwait(false);
+
+                _messagesToSend.TryDequeue(out SystemHealthTelemetry systemHealth);
+                if (systemHealth == null)
+                {
+                    continue;
+                }
+
+                if (s_deviceClients != null)
+                {
+                    foreach (KeyValuePair<string, IotHubDeviceClient> entry in s_deviceClients)
+                    {
+                        string deviceId = entry.Key;
+                        IotHubDeviceClient deviceClient = entry.Value;
+
+                        var telemetryObject = new DeviceTelemetry()
+                        {
+                            DeviceId = deviceId,
+                            SystemHealth = systemHealth,
+                        };
+
+                        var message = new TelemetryMessage(telemetryObject);
+
+                        foreach (KeyValuePair<string, string> prop in TelemetryUserProperties)
+                        {
+                            message.Properties.TryAdd(prop.Key, prop.Value);
+                        }
+
+                        try
+                        {
+                            _logger.Trace($"Sending a telemetry message from the device with Id [{deviceId}].", TraceSeverity.Information);
```

Comment:
You should add device-specific loggers that include their device Id and SDK version as custom properties.